### PR TITLE
Add test-run command

### PR DESCRIPTION
**Pull request by elyezer**
The test-run command parses a jUnit XML file and create a Test Run based
on the results present on the file.

### CODE REVIEW

File: betelgeuse.py [hunk 1]
```diff
@@ -28,7 +39,40 @@ def parse_requirement_name(test_case_id):
...
 
-@click.command()
+def parse_junit(path):
+    """Parse a jUnit XML file.
+
+    :param str path: Path to the jUnit XML file.
+    :return: A list of dicts with information about every test
+        case result.
+    """
+    root = ElementTree.parse(path).getroot()
```
Comment (omaciel) [inline betelgeuse.py:74]: How about handling errors when you cannot parse the `path` provided?

File: betelgeuse.py [hunk 2]
```diff
@@ -28,7 +39,40 @@ def parse_requirement_name(test_case_id):
...
+    result = []
+    for testcase in root.iter('testcase'):
+        data = testcase.attrib
+        element = testcase.find('error')
+        if element is not None:
```
Comment (omaciel) [inline betelgeuse.py]: a lot of this code is repeated... can't we adopt a `for status in ['error', 'failure', 'skipped']` and avoid the repetition?

File: betelgeuse.py [hunk 3]
```diff
@@ -123,3 +165,56 @@ def cli(path, collect_only, project):
...
+@cli.command('test-run')
+@click.option(
+    '--path',
+    default='junit-results.xml',
+    help='Path to the jUnit XML file.',
+    type=click.Path(exists=True, dir_okay=False),
+)
+@click.option(
+    '--test-run-id',
+    default='test-run-{}'.format(time.time()),
```
Comment (omaciel) [inline betelgeuse.py]: How about adding an index of `0` in `'test-run-{}'`?

File: betelgeuse.py [hunk 4]
```diff
@@ -123,3 +165,56 @@ def cli(path, collect_only, project):
...
                         test.docstring if test.docstring else '')
                     test_case.update()
+
+
+@cli.command('test-run')
+@click.option(
+    '--path',
+    default='junit-results.xml',
+    help='Path to the jUnit XML file.',
+    type=click.Path(exists=True, dir_okay=False),
```
Comment (omaciel) [inline betelgeuse.py:197]: Does `exists=True` perform a verification step?

File: betelgeuse.py [hunk 5]
```diff
@@ -123,3 +165,56 @@ def cli(path, collect_only, project):
...
+    help='User that is executing the Test Run.',
+)
+@click.argument('project')
+def test_run(path, test_run_id, test_template_id, user, project):
+    """Execute a test run based on jUnit XML file."""
+    results = parse_junit(path)
+    test_run = TestRun.create(project, test_run_id, test_template_id)
+
+    for result in results:
+        test_case_id = '{}.{}'.format(result['classname'], result['name'])
```
Comment (omaciel) [inline betelgeuse.py]: Please add indices for all string manipulations?

File: betelgeuse.py (hunk 2 above)
Comment (sthirugn) [inline betelgeuse.py]: :+1:

Comment (sthirugn) [inline betelgeuse.py]: also please add some documentation on what you are trying to achieve here.

File: betelgeuse.py (hunk 1 above)
Comment (elyezer) [inline betelgeuse.py:74]: What kind of errors? Click will ensure a valid file path.

File: betelgeuse.py (hunk 3 above)
Comment (elyezer) [inline betelgeuse.py]: Supporting python 2.7+ so no need to add.

File: betelgeuse.py (hunk 4 above)
Comment (elyezer) [inline betelgeuse.py:197]: Yes, click ensures that it is a valid path and is is a file.

File: betelgeuse.py (hunk 5 above)
Comment (elyezer) [inline betelgeuse.py]: We are supporting python2.7+

**Comment by sthirugn** [comment]
ACK

File: betelgeuse.py (hunk 1 above)
Comment (omaciel) [inline betelgeuse.py:74]: Ahhh that was what I asked further down. Good for me.

File: betelgeuse.py (hunk 3 above)
Comment (omaciel) [inline betelgeuse.py]: Explicit is better than implicit. NACK

File: betelgeuse.py (hunk 5 above)
Comment (omaciel) [inline betelgeuse.py]: See my previous comment. NACK if not using indices

**Comment by omaciel** [comment]
NACK until indices are used. Never say never :)

File: betelgeuse.py (hunk 3 above)
Comment (elyezer) [inline betelgeuse.py]: Not needing indices still an explicit python feature :smile:

**Comment by omaciel** [comment]
ACK

**Pull request merged**